src/edu/csus/ecs/pc2/clics/API202306/AwardService.java

```diff
@@ -0,0 +1,260 @@
+                List<CLICSAward> genericCLICSAwards = CLICSAwardUtilities.createAwardsList(model, obeyFreeze);
+                
+                if(genericCLICSAwards != null) {
+                    // Convert each award to this API's idea of an award
+                    for(CLICSAward award: genericCLICSAwards) {
+                        if(awardId.equals(award.getId())) {
+                            return Response.ok(new CLICS202306Award(awardId, award.getCitation(), award.getTeam_ids()).toJSON(), MediaType.APPLICATION_JSON).build();                            
+                        }
+                    }
+                }
+            } catch(Exception e) {
+                // some error occurred and we really don't care: it's an error, just log it and return a bad reply
+                controller.getLog().log(Log.WARNING, "can not get awards list", e);
+                return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for awards " + e.getMessage()).build();
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    /**
+     * Post a new award.
+     * 
+     * @param servletRequest details of request
+     * @param sc requesting user's authorization info
+     * @param contestId The contest
+     * @param jsonInputString citation and team_ids json for the new award
+     * @return json for the award, including the id
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{contestId}/awards")
+    public Response addNewAward(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, String jsonInputString) {
+        
+        if(sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) == false) {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        
+        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
+    }
+    
+    /**
+     * Put updates an existing award
+     * 
+     * @param servletRequest details of request
+     * @param sc requesting user's authorization info
+     * @param contestId The contest
+     * @param jsonInputString citation and team_ids json for the new award
+     * @return json for the award, including the id
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{contestId}/awards/{awardId}")
+    public Response setAwardInfo(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("awardId") String awardId, String jsonInputString) {
+        
+        if(sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) == false) {
+            return Response.status(Response.Status.FORBIDDEN).build();
+        }
+        
+        return Response.status(Response.Status.NOT_IMPLEMENTED).build();
```

Comment:
It seems like this condition should be logged...

Reply:
See above.  We can just say at this point, that if we decide to log these, we have to go through all the services and check every non success return code and add logging for it.  To be discussed at the meeting.